examples/all-clusters-app/all-clusters-common/src/operational-state-delegate-impl.cpp

```diff
@@ -0,0 +1,400 @@
+        des->operationalStateID = src->operationalStateID;
+
+        if (des == nullptr)
+        {
+            err = CHIP_ERROR_NO_MEMORY;
+            ExitNow();
+        }
+
+        if (head == nullptr)
+        {
+            head = des;
+        }
+        else
+        {
+            GenericOperationalStateList * pList = head;
+            while (pList->next != nullptr)
+            {
+                pList = pList->next;
+            }
+            pList->next = des;
+        }
+        src++;
+    }
+    size                  = i;
+    *operationalStateList = head;
+    return CHIP_NO_ERROR;
+exit:
+    ReleaseOperationalStateList(head);
+    return err;
+}
+
+void OperationalStateDelegate::ReleaseOperationalStateList(GenericOperationalStateList * operationalStateList)
+{
+    while (operationalStateList)
+    {
+        GenericOperationalStateList * del = operationalStateList;
+        operationalStateList              = operationalStateList->next;
+        chip::Platform::Delete(del);
+    }
+}
+
+CHIP_ERROR OperationalStateDelegate::GetOperationalPhaseList(GenericOperationalPhaseList ** operationalPhaseList, size_t & size)
+{
+    CHIP_ERROR err                     = CHIP_ERROR_NO_MEMORY;
+    size                               = 0;
+    size_t i                           = 0;
+    size_t phaseListNumOfItems         = 0;
+    GenericOperationalPhaseList * head = nullptr;
+
+    if (!operationalPhaseList)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    const GenericOperationalPhase * src = getGenericPhaseListTable(mEndpointId, mClusterId, phaseListNumOfItems);
+    if (!src || !phaseListNumOfItems)
+    {
+        ChipLogError(Zcl, "Unable to find Phase List for [ep=%d],[cid=%d]", mEndpointId, mClusterId);
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    for (i = 0; i < phaseListNumOfItems; i++)
+    {
+        GenericOperationalPhaseList * des = nullptr;
+        if (src->isNullable())
+        {
+            des = chip::Platform::New<GenericOperationalPhaseList>();
+        }
+        else
+        {
+            des = chip::Platform::New<GenericOperationalPhaseList>(src->PhaseName, sizeof(src->PhaseName));
+        }
+
+        if (des == nullptr)
+        {
+            err = CHIP_ERROR_NO_MEMORY;
+            ExitNow();
+        }
+
+        if (head == nullptr)
+        {
+            head = des;
+        }
+        else
+        {
+            GenericOperationalPhaseList * pList = head;
+            while (pList->next != nullptr)
+            {
+                pList = pList->next;
+            }
+            pList->next = des;
+        }
+        src++;
+    }
+    size                  = i;
+    *operationalPhaseList = head;
+    return CHIP_NO_ERROR;
+exit:
+    ReleaseOperationalPhaseList(head);
+    return err;
+}
+
+void OperationalStateDelegate::ReleaseOperationalPhaseList(GenericOperationalPhaseList * operationalPhaseList)
+{
+    while (operationalPhaseList)
```

Comment:
Don't we have a maximum on the list for operational state ?, or it is unlimited i don't remember the spec here.

Reply:
I have remove this api, the spec doesn't define the maximum of  the list for operational state.